experimental/traffic-portal/src/app/core/currentuser/currentuser.component.html

```diff
@@ -11,67 +11,71 @@
-	</mat-form-field>
-	<mat-form-field style="grid-area: j;">
-		<mat-label>Postal Code</mat-label>
-		<input matInput [(ngModel)]="editUser.postalCode" name="postalCode" type="text"/>
-	</mat-form-field>
-	<button mat-raised-button style="grid-area: k;">Submit</button>
-	<button mat-raised-button color="accent" style="grid-area: l;" type="button" (click)="updatePassword()">Update Password</button>
-	<button type="button" mat-raised-button color="warn" (click)="cancelEdit()" style="grid-area: m;">Cancel</button>
-</form>
+	<form *ngIf="editMode && editUser" ngNativeValidate (submit)="submitEdit($event)">
+		<mat-form-field style="grid-area: a-start;">
+			<mat-label>Username</mat-label>
+			<input matInput [(ngModel)]="editUser.username" name="username" type="text" required/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: z;">
+			<mat-label>Full Name</mat-label>
+			<input matInput [(ngModel)]="editUser.fullName" name="fullName" type="text" required/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: b;">
+			<mat-label>Email</mat-label>
+			<input matInput [(ngModel)]="editUser.email" name="email" type="email"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: c;">
+			<mat-label>Telephone</mat-label>
+			<input matInput [(ngModel)]="editUser.phoneNumber" name="phoneNumber" type="tel"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: d;">
+			<mat-label>Address Line 1</mat-label>
+			<input matInput [(ngModel)]="editUser.addressLine1" name="addressLine1" type="text"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: e;">
+			<mat-label>Address Line 2</mat-label>
+			<input matInput [(ngModel)]="editUser.addressLine2" name="addressLine2" type="text"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: f;">
+			<mat-label>Company</mat-label>
+			<input matInput [(ngModel)]="editUser.company" name="company" type="text"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: g;">
+			<mat-label>City</mat-label>
+			<input matInput [(ngModel)]="editUser.city" name="city" type="text"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: h;">
+			<mat-label>State or Province</mat-label>
+			<input matInput [(ngModel)]="editUser.stateOrProvince" name="stateOrProvince" type="text"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: i;">
+			<mat-label>Country</mat-label>
+			<input matInput [(ngModel)]="editUser.country" name="country" type="text"/>
+		</mat-form-field>
+		<mat-form-field style="grid-area: j;">
+			<mat-label>Postal Code</mat-label>
+			<input matInput [(ngModel)]="editUser.postalCode" name="postalCode" type="text"/>
+		</mat-form-field>
+		<button mat-raised-button style="grid-area: k;">Submit</button>
+		<button mat-raised-button color="accent" style="grid-area: l;" type="button" (click)="updatePassword()">Update Password</button>
+		<button type="button" mat-raised-button color="warn" (click)="cancelEdit()" style="grid-area: m;">Cancel</button>
```

Comment:
should these be in the card actions?

Reply:
I don't think so, at least not without restructuring the whole form due to the fact adding `mat-card-actions` really messes with the grid layout

Reply:
I don't think it would be hard, but this is your crusade so I'll defer to you.